--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.16)
 project(fptn)

+option(USE_BORINGSSL "Use BoringSSL" OFF)
+
 if (USE_CONAN)
     message(FATAL_ERROR "Conan is disabled for OpenWrt builds")
 endif()
@@ -20,7 +22,11 @@
 endif()

-if (USE_BORINGSSL)
-    find_package(BoringSSL REQUIRED)
-endif()
+if (NOT USE_BORINGSSL)
+    find_package(OpenSSL REQUIRED)
+endif()
